src/zenml/utils/pipeline_docker_image_builder.py

```diff
@@ -356,7 +367,10 @@ def _add_requirements_files(
             build_context.add_file(source=file_content, destination=filename)
             requirements_file_names.append(filename)
 
-        return requirements_file_names
+        all_requirements = "\n".join(
+            file_content for _, file_content in requirements_files
+        )
+        return requirements_file_names, all_requirements or None
```

Comment:
How about we move this logic to `build_docker_image` and keep this function unchanged?

Reply:
Yeah I actually refactored it quite a bit now, let me know if it still needs any changes

Reply:
LGTM